File: real-estate-bidding-system-api/src/main/java/org/universe/realestatebiddingsystem/estates/estate/model/view/EstateViewModel.java
package org.universe.realestatebiddingsystem.estates.estate.model.view;

import lombok.Data;
import org.universe.realestatebiddingsystem.estates.peculiarity.model.view.PeculiarityViewModel;
import org.universe.realestatebiddingsystem.user.model.view.UserViewModel;

import java.util.List;

@Data
public class EstateViewModel {
    private Long id;
    private String action;
    private String type;
    private String additionalInfo;
    private String city;
    private Double area;
    private Double price;
    private ImageViewModel coverImage;
    private List<ImageViewModel> images;
    //    private String firstImage;
//    private String secondImage;
//    private String thirdImage;
    private UserViewModel author;
    private Double lastBid;
    private PeculiarityViewModel[] peculiarities;
}

File: real-estate-bidding-system-client/src/app/core/model/response/estate/estate-view.model.ts
import {PeculiarityViewModel} from '../../view/peculiarity/peculiarity-view.model';
import {ImageViewModel} from '../../view/image/image-view.model';
import {UserViewModel} from '../../view/user/user-view.model';

export class EstateViewModel {
  constructor(
    public id: number,
    public action: string,
    public type: string,
    public city: string,
    public area: number,
    public price: number,
    public coverImage: ImageViewModel,
    public images: ImageViewModel[],
    public additionalInfo: string,
    public author: UserViewModel,
    public peculiarities: PeculiarityViewModel[],
    public lastBid: number
  ) {
  }
}
